### PR TITLE
cephfs-top: fixing the -d [--delay] option in cephfs-top

### PR DESCRIPTION
**Pull request by neesingh-rh**
`cephfs-top -d [--delay]` excepts for float values because
`curses.halfdelay()` accepts only integer values.

Fixes: https://tracker.ceph.com/issues/59188

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Review by anthonyeleven — COMMENTED** [review]
docs lgtm

File: src/tools/cephfs/top/cephfs-top [hunk 1]
```diff
@@ -1191,6 +1191,10 @@ if __name__ == '__main__':
             raise argparse.ArgumentTypeError(
                 'Refresh interval should be greater than or equal to'
                 f' {MIN_REFRESH_INTERVAL}')
+        elif isinstance(x, int):
+            value = x * 10
+        else:
+            value = int(round(x, 1) * 10)
```
Comment (joscollin) [inline src/tools/cephfs/top/cephfs-top:1197]: What's the point of honouring fractional seconds then? You are rounding it to the nearest integer anyway. So why don't we just accept only integer input for `--delay` option, which fixes the issue. curses.halfdelay only accepts integer arguments between 1 and 255. @vshankar Any reason we honour fractional seconds?

I'm attaching the code I've done to fix the same.

[fix.txt](https://github.com/ceph/ceph/files/11109609/fix.txt)

Comment (vshankar) [reply src/tools/cephfs/top/cephfs-top:1197]: > What's the point of honouring fractional seconds then? You are rounding it to the nearest integer anyway. So why don't we just accept only integer input for `--delay` option, which fixes the issue. curses.halfdelay only accepts integer arguments between 1 and 255. @vshankar Any reason we honour fractional seconds?

Nothing specific - just that users can choose 1.5s as delay. top(1) supports that.

> 
> I'm attaching the code I've done to fix the same.
> 
> [fix.txt](https://github.com/ceph/ceph/files/11109609/fix.txt)

Comment (joscollin) [reply src/tools/cephfs/top/cephfs-top:1197]: There won't be a noticeable difference in refresh delay having a fractional second with it's nearest integer. So I think it's better avoid it, if there's no specific need.

Comment (vshankar) [reply src/tools/cephfs/top/cephfs-top:1197]: Understood. Thanks for clarifying.

Comment (neesingh-rh) [reply src/tools/cephfs/top/cephfs-top:1197]: for eg: if user enters delay= 12.43, then `int(round(delay, 1) * 10)` will be `124` and `curses.halfdelay()` accepts 124 and waits for `124/10 = 12.4s` which was expected by the user.

Comment (neesingh-rh) [reply src/tools/cephfs/top/cephfs-top:1197]: Its correct that there's not going to be much difference by having the decimal places in delay, but as discussed by @vshankar to keep the fractional seconds supportability, I went ahead this way.

Comment (vshankar) [reply src/tools/cephfs/top/cephfs-top:1197]: I could have misread the docs and misunderstood that halfdelay accepts fractional values rounded upto one decimal place. I'm fine with integer only values if that's not the case.

Comment (joscollin) [reply src/tools/cephfs/top/cephfs-top:1197]: No, it accepts only integer values. There's no reason to make the user think that halfdelay() processes fractional values.

File: src/tools/cephfs/top/cephfs-top [hunk 2]
```diff
@@ -271,7 +269,7 @@ class FSTop(FSTopBase):
         self.client_name = args.id
         self.cluster_name = args.cluster
         self.conffile = args.conffile
-        self.refresh_interval_secs = args.delay
+        self.refresh_interval_secs = args.delay * 10
```
Comment (joscollin) [inline src/tools/cephfs/top/cephfs-top:272]: There's no need to increase delay in functions like `set_key`, `choose_field` etc. A minimal delay is set there just to make the input mechanism to work correctly. But you can multiply in `run_display` and `run_all_display` (within range 1 - 255), where the user expects a refresh delay.

**Review by joscollin — CHANGES_REQUESTED** [review]

**Review by joscollin — APPROVED** [review]

File: src/tools/cephfs/top/cephfs-top [hunk 3]
```diff
@@ -1202,9 +1192,9 @@ if __name__ == '__main__':
                         help='Path to cluster configuration file')
     parser.add_argument('--selftest', dest='selftest', action='store_true',
                         help='Run in selftest mode')
-    parser.add_argument('-d', '--delay', nargs='?',
+    parser.add_argument('-d', '--delay', dest='delay', choices=range(1, 26),
```
Comment (vshankar) [inline src/tools/cephfs/top/cephfs-top:1195]: Does this mean values between 1 and 26 are allowed? (I haven't looked up what choices does).

Comment (neesingh-rh) [reply src/tools/cephfs/top/cephfs-top:1195]: Values from 1 to 25 are allowed, as curses.halfdelay() limits to 255 .

Comment (joscollin) [reply src/tools/cephfs/top/cephfs-top:1195]: @neesingh-rh exactly

**Review by vshankar — APPROVED** [review]

**Pull request merged**